### ISSUE TITLE
rustdoc hits windows path limit (260 characters)

### ISSUE THREAD

**Issue by HK416-is-all-you-need** [rust-lang/rust#91623]
subj.
Error looks like:
```
error: "<path>": The system cannot find the path specified. (os error 3)
```
Where `path` is string over 260 characters

To fix it rustdoc would need to prefix all paths with `\\?\` on windows
Not sure if this issue has been risen before or not, is it acceptable to fix it?
```
$ rustc --version
rustc 1.57.0 (f1edd0429 2021-11-29)
$ rustdoc --version
rustdoc 1.57.0 (f1edd0429 2021-11-29)
```

**Comment by ChrisDenton** [comment]
Can you try nightly rustdoc?

```plain
rustup install nightly
rustdoc +nightly <path>
```

**Comment by HK416-is-all-you-need** [comment]
@ChrisDenton Just in case I re-checked with latest available nightly there is actually slightly different issue:
```
The filename, directory name, or volume label syntax is incorrect. (os error 123)
```

I'm not sure what exactly changed but could be there are some wrong slashes inside path?
Unfortunately I cannot be sure as error output contains regular windows path like "C:\<path>"

**Comment by HK416-is-all-you-need** [comment]
@ChrisDenton Ah my bad, I used enormously large struct name to have simplified test case.
And while stable release failed with code 3, I suspect winapi doesn't like big file names hence it failed on nightly too.

I retried my original use case where generated structs names are not too large and successfully build docs on nightly, while stable returned error code 3.
As this is current latest nightly we can consider issue fixed.
